astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +187,59 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(
+    PyUnitListProxy *a,
+    PyUnitListProxy *b,
+    int op) {
+
+  int equal;
+  int status;
+  struct PyUnitListProxy *ax;
+  struct PyUnitListProxy *bx;
+   
+  
+  if ((op == Py_EQ || op == Py_NE) &&
```

Comment:
Could you negate that check and return the `Py_NotImplemented` immediately here? It makes the following (already nested) code easier to understand.